### ISSUE TITLE
Opal Ice Maker Light Level state in HA doesn't update when changed from an Automation, Entity or SmartHQ App

### ISSUE THREAD

**Issue by KHouse75** [simbaja/ha_gehome#132]
When I configure an automation to change the Opal ice maker light level, the ice maker light changes as expected when the automation is run but the opal ice maker state in home assistant does not change.

**Comment by KHouse75** [comment]
I installed the sdk and was able to gather more info.

With gehome-appliance-data running, when I change the light level through home assistant or through the android application , nothing shows up in the debug output but the light level on the Opal Ice Maker does change.  However, when I change the light level by pressing the Opal Ice Maker button, the light level does change and the following is displayed in the debug log.

Is there anything I can do to gather more information to help troubleshoot this issue?

```
2022-12-22 21:20:16,871 DEBUG    Setting ErdCode.OIM_LIGHT_LEVEL to ErdOimLightLevel.DIM
2022-12-22 21:20:16,871 DEBUG    Appliance state change detected in GeAppliance(D828C9D9AAB3) (ErdApplianceType.OPAL_ICE_MAKER). Updated keys: ErdCode.OIM_LIGHT_LEVEL
2022-12-22 21:20:26,813 DEBUG    Setting ErdCode.OIM_LIGHT_LEVEL to ErdOimLightLevel.OFF
2022-12-22 21:20:26,814 DEBUG    Appliance state change detected in GeAppliance(D828C9D9AAB3) (ErdApplianceType.OPAL_ICE_MAKER). Updated keys: ErdCode.OIM_LIGHT_LEVEL
```

**Comment by simbaja** [comment]
@knobunc / @mbcomer, does your device exhibit this issue too or are there perhaps multiple models out there?

**Comment by mbcomer** [comment]
> @knobunc / @mbcomer, does your device exhibit this issue too or are there perhaps multiple models out there?

Unfortunately I have my ice maker shut down for the winter - we don't do a lot of icy drinks and such when it's cold outside. I can certainly check once I clean it up and bring it back online.

**Comment by N0iSEA** [comment]
I have the Opal Icemaker 2.0. The problem seems to be that this is supposed to be one button that changes in the following sequence each time its pressed. Off - Bright - Dim - Off - Bright - Dim etc.

However it shows up as 3 seperate selections in home assistant. To make it work you need to select them in that order to change them. So if the machine light is on Bright and you want it on off, you must first select Dim, and then next you should select off. Or if it is off and you want it on dim, you first select bright and then you can select off.

Hopefully someone can fix this to be only a single button that toggles through the lights in that order.


(By the way the setting I am calling bright is called dim in HA - i just didn’t want to make it confusing by saying “On” since “Dim” is also on.)

**Comment by simbaja** [comment]
Interesting, does the SmartHQ app just have something that toggles, or can it be set directly?

**Comment by N0iSEA** [comment]
The SmartHQ app has 3 separate buttons similar to the HA integration. But
It works in the app how you would expect - you can choose them in any order
you wish, unlike in HA, which requires a specific order.

However, the button on the actual machine works like the HA integration
(its one button and it cycles in a specific order off-bright-dim-off-etc)

**Comment by simbaja** [comment]
Any chance you can go to simbaja/gehome repository and run that application data command?  If you run it while hitting the buttons in the SmartHQ app, we might be able to see if it's doing something differently and figure out how to enable that functionality.

**Comment by N0iSEA** [comment]
I will give it a try after work.

**Comment by N0iSEA** [comment]
When i look in the HA Logbook for OIM Light Level, which is the entity that controls the light through simbaja/gehome, nothing happens when I hit the buttons in the SmartHQ app, but the lights do change as expected on the ice maker based on the button I select.

However, when I hit the actual button on the Opal Ice Maker machine, the activity is listed in the logbook.

Similarly, When I make the selections in HA the selections show up in the logbook.

I am not a programmer (just a clumsy tinkerer), so take this with a grain of salt, but think that means that the app is using different code to trigger the light change than the simbaja/gehome, which uses the same code as the physical button?

HOWEVER, all that being said... I am not experiencing the same issue today where the buttons will only work when pressed in the correct sequence. I was pleasantly surprised that I couldn't replicate that situation at all today. I don't know what has changed since yesterday, but I tested it dozens of times yesterday and it was only working when I selected them in the sequence, otherwise no change (i swear I was sober when i was doing it too!).

So I don't know what to say. it seems to be working as intended for me personally now.
